syntax = "proto3";

package misollar_service;

option go_package = "genproto/misollar_service";

service MisollarService {
    rpc get_misollar_query(MisollarRequest) returns (MisollarResponse);
    rpc get_misollar_complete(MisollarRequest) returns (CompleteResponse);
}

message CompleteResponse {
    repeated string src_phrase = 1;
}

message MisollarRequest {
    string src_phrase = 1;
    string src_lang = 2;
    string tgt_lang = 3;
}

message Span {
    int32 start = 1;
    int32 end = 2;
}

message Document {
    string id = 1;
    string sent_id = 2;
    string src_lang = 3;
    string src_phrase = 4;
    string src_sentence = 5;
    repeated Span src_spans = 6;
    string tgt_lang = 7;
    string tgt_phrase = 8;
    string tgt_sentence = 9;
    repeated Span tgt_spans = 10;
    string source = 11;
}

message MisollarResponse {
    repeated Document hits = 1;
}
